File: app/src/main/java/com/github/premnirmal/ticker/model/SerializableDataPoint.java
package com.github.premnirmal.ticker.model;

import com.jjoe64.graphview.series.DataPoint;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by premnirmal on 2/20/15.
 */
public class SerializableDataPoint extends DataPoint implements Serializable {

    public SerializableDataPoint(double x, double y) {
        super(x, y);
    }

    public SerializableDataPoint(Date x, double y) {
        super(x, y);
    }
}

File: app/src/main/kotlin/com/github/premnirmal/ticker/model/SerializableDataPoint.kt
package com.github.premnirmal.ticker.model

import android.os.Parcel
import android.os.Parcelable
import com.github.mikephil.charting.data.Entry
import com.github.premnirmal.ticker.network.data.historicaldata.HistoryQuote
import java.io.Serializable

/**
 * Created by premnirmal on 2/28/16.
 */
class SerializableDataPoint : Entry, Serializable {

  constructor(y: Float, x: Int) : super(y, x)

  constructor(y: Float, x: Int, data: HistoryQuote) : super(y, x, data)

  constructor(source: Parcel) : super(source)

  fun getQuote(): HistoryQuote {
    return (data as HistoryQuote)
  }

  companion object {

    private val serialVersionUID = 42L

    val CREATOR: Parcelable.Creator<SerializableDataPoint> = object : Parcelable.Creator<SerializableDataPoint> {
      override fun createFromParcel(source: Parcel): SerializableDataPoint {
        return SerializableDataPoint(source)
      }

      override fun newArray(size: Int): Array<SerializableDataPoint?> {
        return arrayOfNulls(size)
      }
    }
  }
}